src/pages/index.js

```diff
@@ -52,28 +100,54 @@ cardSection.renderItems();
 const editProfilePopup = new PopupWithForm(
   '#profile-edit-modal',
   (formData) => {
-    userInfo.setUserInfo({
-      name: formData.title,
-      description: formData.description
-    });
+    api.setUserInfo(formData.title, formData.description)
+      .then(updatedUserData => {
+        userInfo.setUserInfo({
+          name: updatedUserData.name,
+          description: updatedUserData.about,
+          avatar: updatedUserData.avatar 
+        });
+        editProfilePopup.close();
+      })
+      .catch(err => console.error(err));
   }
 );
 editProfilePopup.setEventListeners();
 
-const addCardPopup = new PopupWithForm(
-  '#profile-add-card',
-  (formData) => {
-    const newCard = createCard({
-      name: formData.title,
-      link: formData.url
+const addCardPopup = new PopupWithForm('#profile-add-card', (formData) => {
+  api.addNewCard(formData.title, formData.url)
+    .then((newCardData) => {
+      const newCardElement = createCard(newCardData);
+      cardSection.addItem(newCardElement);
+      
+      addCardPopup.close();
+      addCardFormValidator.disableButton();
+    })
+    .catch((err) => {
+      console.error('Error adding card:', err);
     });
-    cardSection.addItem(newCard);
-
-    addCardFormValidator.disableButton();
-  }
-);
+});
 addCardPopup.setEventListeners();
 
+
+const avatarPopup = new PopupWithForm('#avatar-edit-modal', (formData) => {
+  api.setAvatar(formData.avatar)
+    .then(updatedUser => {
+      userInfo.setUserInfo({
+        name: updatedUser.name,
+        description: updatedUser.about,
+        avatar: updatedUser.avatar
+      });
+      avatarPopup.close();
+    })
+    .catch(err => console.error(err));
+});
+
+avatarPopup.setEventListeners();
+avatarContainer.addEventListener('click', () => {
+   avatarPopup.open();
+  });
+
```

Comment:
* Please, create the `avatar` form **validator**, otherwise vilidation of the `avatar` form doesn't work